README.md

```diff
@@ -2,140 +2,167 @@
-![img.png](https://raw.githubusercontent.com/run-house/runhouse/main/docs/assets/img_1.png)
+gpu.up_if_not()  # Optional, as it's called automatically when you use the cluster
+```
 
-It wraps industry-standard tooling like Ray and the Cloud SDKs (boto, gsutil, etc. via [SkyPilot](https://github.com/skypilot-org/skypilot/))
-to give you production-quality features like queuing, distributed, async, logging,
-low latency, hardware efficiency, auto-launching, and auto-termination out of the box.
+There are lots of things you can send to a cluster. For example, a folder, from local or cloud storage (these work 
+in any direction, so you can send folders arbitrarily between local, cloud storage, and cluster storage):
 
-### 👩‍💻 Enough chitchat, just show me the code
+```python
+my_cluster_folder = rh.folder(path="./my_folder").to(gpu, path="~/my_folder")
+my_s3_folder = my_cluster_folder.to("s3", path="my_bucket/my_folder")
+my_local_folder = my_s3_folder.to("here")
+```
 
-Here is **all the code you need** to stand up a stable diffusion inference service on
-a fresh cloud GPU.
+You can send a function to the cluster, including the environment in which the function will live, which is actually
+set up in its own Ray process. You can send it to an existing env, or create a new one on the fly. Like the folder 
+above, the function object which is returned from `.to` is a proxy to the remote function. When you call it a 
+lightweight request is sent to the cluster's Runhouse HTTP server to execute the function with the given inputs and 
+returns the results. Note that the function is not serialized, but rather imported on the cluster after the local 
+working directory (`"./"`, by default the git root) is sent up.
 
 
 ```python
-import runhouse as rh
 from diffusers import StableDiffusionPipeline
 
 def sd_generate(prompt):
     model = StableDiffusionPipeline.from_pretrained("stabilityai/stable-diffusion-2-base").to("cuda")
-    return model(prompt).images[0]
-
-gpu = rh.ondemand_cluster(name="my-a100", instance_type="A100:1").up_if_not()
-sd_generate = rh.function(sd_generate).to(gpu, env=["torch", "diffusers"])
+    return model(prompt).images
 
-# the following runs on our remote A100 gpu
-sd_generate("An oil painting of Keanu Reeves eating a sandwich.").show()
-sd_generate.save(name="sd_generate")
+if __name__ == "__main__":
+    remote_sd_generate = rh.function(sd_generate).to(gpu, env=["./", "torch", "diffusers"])
+    imgs = remote_sd_generate("A hot dog made out of matcha.")
+    imgs[0].show()
 ```
 
-On the data side, sync folders, tables, or blobs between local, clusters, and file storage. All
-this is done without bouncing off the laptop.
+The above env list is a shorthand, but you can create quite elaborate envs (even quite a bit more elaborate than the 
+example below). Local folders are sent (rsynced) up as needed, and the environment setup is cached so it only reruns 
```

Comment:
Nit, remove this sentence "Local folders...". You can just talk about the more elaborate envs, no need to discuss what's under the hood if it just works well